test/Microsoft.DocAsCode.Build.Engine.Tests/TemplateManagerUnitTest.cs

```diff
@@ -573,6 +653,80 @@ public void TestLiquidTemplateProcessSingleTemplateWithDependenciesShouldWork()
         [Trait("Related", "TemplateProcessor")]
         [Trait("Related", "Liquid")]
         [Fact]
+        public void TestLiquidTemplateWithMasterPageShouldWork()
+        {
+            // 1. Prepare template
+            var templateName = "Subfolder/LiquidWithMasterPage";
+            string template = @"
+{% master _layout/master.html -%}
+{% master _layout/master.html -%}
+{% for item in model -%}
+{{ item.name }}
+{% endfor -%}
+{% include partial1 -%}
+";
+            string partial1 = @"partial 1:
+{% include partial2 -%}";
+            string partial2 = @"partial 2:
+{% for item in model -%}
+{{ item.name }}
+{% endfor -%}
+";
+
+            string master = @"
+Hello Master
+{% include partial1 -%}
+{%- body %}
```

Comment:
minor: update pair to {%- x -%} or {% x %}? starts with %- and ends without -% is not align?